torch/_inductor/codegen/rocm/ck_universal_gemm_template.py

```diff
@@ -0,0 +1,370 @@
+import logging
+import random
+from typing import List, Optional
+
+import sympy
+
+import torch
+from torch._inductor import config
+from torch._inductor.codegen.rocm.ck_library import gen_ops_library, gen_ops_preselected
+from torch._inductor.codegen.rocm.ck_template import CKTemplate
+from torch._inductor.codegen.rocm.ck_universal_gemm_op import CKGemmOperation
+from torch._inductor.codegen.rocm.rocm_kernel import ROCmTemplateKernel
+from torch._inductor.ir import Buffer, Layout
+
+log = logging.getLogger(__name__)
+
+from ...utils import IndentedBuffer
+
+
+def is_static_int(number):
+    return isinstance(number, (int, sympy.Integer))
+
+
+def torch_layout_to_ck_layout(torch_layout):
+    if torch_layout.stride[-1] == 1:
+        return "Row"
+    elif torch_layout.stride[-2] == 1:
+        return "Col"
+    else:
+        return None
+
+
+class CKGemmTemplate(CKTemplate):
+    # the JINJA template for rendering CK Universal GEMMs
+    gemm_template = r"""{{version_comment}}
+    {{headers}}
+    {{globals}}
+    {{instance_definition}}
+    extern "C" {
+    {{kernel_definition}} {
+        auto gemm = {{instance_type}} {};
+        auto invoker = gemm.MakeInvoker();
+
+        constexpr auto M = {{M}};
+        constexpr auto N = {{N}};
+        constexpr auto K = {{K}};
+        constexpr auto StrideA = std::is_same_v<{{a_layout}}, Row> ? K : M;
+        constexpr auto StrideB = std::is_same_v<{{b_layout}}, Row> ? N : K;
+        constexpr auto StrideC = std::is_same_v<{{c_layout}}, Row> ? N : M;
+        constexpr auto KBatch = 1; // split k into batches
+
+        auto argument = gemm.MakeArgument(
+            reinterpret_cast<const {{a_element_dtype}}*>(X),
+            reinterpret_cast<const {{b_element_dtype}}*>(W),
+            reinterpret_cast<{{c_element_dtype}}*>(Y),
+            M,
+            N,
+            K,
+            StrideA,
+            StrideB,
+            StrideC,
+            KBatch,
+            {{a_elementwise_op}} {},
+            {{b_elementwise_op}} {},
+            {{c_elementwise_op}} {}
+        );
+        if (!gemm.IsSupportedArgument(argument)) {
+            // magic spell to assign +inf to benchmarking time in select_algorithm.py:1052 (1/2)
```

Comment:
is there a better way of doing this 🤔

Reply:
hmm it looks like an outdated code comment. I think the algorithm choice just ignores the choices which raised an error during benchmarking now. I'll clean up